### PR TITLE
v6 gateway sample

### PR DESCRIPTION
**Pull request by janovesk**

### CODE REVIEW

**Comment by janovesk** [comment]
Need another pair of eyes on this. If good, pull it. Ping @Particular/documentation

File: samples/gateway/Version_6/RemoteSite/App.config [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <section name="GatewayConfig" type="NServiceBus.Config.GatewayConfig, NServiceBus.Gateway"/>
+  </configSections>
+  <!--startcode RemoteSiteGatewayConfig-->
+  <GatewayConfig TransactionTimeout="00:10:00">
+    <Channels>
+      <Channel Address="http://localhost:25899/RemoteSite/" ChannelType="Http" Default="true"/>
+    </Channels>
+  </GatewayConfig>
+  <!--endcode -->
+<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2"/></startup></configuration>
```
Comment (SimonCropp) [inline samples/gateway/Version_6/RemoteSite/App.config]: Remoce the startup

File: samples/gateway/Version_6/Headquarters/App.config [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <GatewayConfig>
+    <Sites>
+      <Site Key="RemoteSite" Address="http://localhost:25899/RemoteSite/" ChannelType="Http"/>
+    </Sites>
+    <Channels>
+      <Channel Address="http://localhost:25899/Headquarters/" ChannelType="Http"/>
+    </Channels>
+  </GatewayConfig>
+  <!--endcode -->
+<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2"/></startup></configuration>
```
Comment (SimonCropp) [inline samples/gateway/Version_6/Headquarters/App.config]: Remoce the startup

File: samples/gateway/Version_6/Headquarters/Program.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+class Program
+{
+    static void Main()
+    {
+        AsyncMain().GetAwaiter().GetResult();
+    }
+
+    static async Task AsyncMain()
+    {
+        EndpointConfiguration configuration = new EndpointConfiguration();
```
Comment (SimonCropp) [inline samples/gateway/Version_6/Headquarters/Program.cs]: Everywhere else we use endpointConfiguration as the variable name

File: samples/gateway/Version_6/Shared/ConfigErrorQueue.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using NServiceBus.Config;
+using NServiceBus.Config.ConfigurationSource;
+
+class ConfigErrorQueue : IProvideConfiguration<MessageForwardingInCaseOfFaultConfig>
```
Comment (SimonCropp) [inline samples/gateway/Version_6/Shared/ConfigErrorQueue.cs]: Use the code api inside the program.cs

File: samples/gateway/Version_6/WebClient/Web.config [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0"?>
+<configuration>
+  <!--
+    For a description of web.config changes see http://go.microsoft.com/fwlink/?LinkId=235367.
```
Comment (SimonCropp) [inline samples/gateway/Version_6/WebClient/Web.config]: Remove the comment

**Comment by SimonCropp** [comment]
Apart from a few minor things this is good

**Comment by janovesk** [comment]
All fixed

**Pull request merged**